storage/scylla_api.c

```diff
@@ -395,31 +399,138 @@ static status_t create_edge_table(CassSession* session) {
                     "address blob,"
                     "hash blob,"
                     "PRIMARY KEY (edge, bundle, address, hash));") != CASS_OK) {
-    ta_log_error("%s\n", "SC_STORAGE_CASSANDRA_QUREY_FAIL");
+    ta_log_error("Create edgeTable fail\n");
+    ret = SC_STORAGE_CASSANDRA_QUREY_FAIL;
+    goto exit;
+  }
+
+exit:
+  return ret;
+}
+
+static status_t create_identity_table(CassSession* session, bool need_drop) {
+  status_t ret = SC_OK;
+  if (need_drop) {
+    if (execute_query(session, "DROP TABLE IF EXISTS identity;") != CASS_OK) {
+      ta_log_error("drop identity table fail\n");
+      ret = SC_STORAGE_CASSANDRA_QUREY_FAIL;
+      goto exit;
+    }
+  }
+  if (execute_query(session,
+                    "CREATE TABLE IF NOT EXISTS identity("
+                    "ID bigint, hash blob, status tinyint, PRIMARY KEY (ID));") != CASS_OK) {
+    ta_log_error("create identity table fail\n");
+    ret = SC_STORAGE_CASSANDRA_QUREY_FAIL;
+    goto exit;
+  }
+  if (execute_query(session, "CREATE INDEX IF NOT EXISTS ON identity(status);") != CASS_OK) {
+    ta_log_error("create identity table index fail\n");
     ret = SC_STORAGE_CASSANDRA_QUREY_FAIL;
     goto exit;
   }
 
 exit:
   return ret;
 }
-status_t init_scylla(CassCluster** cluster, CassSession* session, char* hosts, bool is_need_create_table,
-                     const char* keyspace_name) {
-  if (hosts == NULL) {
-    ta_log_error("%s\n", "SC_TA_NULL");
+
+status_t db_client_service_init(db_client_service_t* service, const char* host) {
+  if (host == NULL || service == NULL) {
+    ta_log_error("Invaild input : NULL pointer\n");
     return SC_TA_NULL;
   }
+  service->session = cass_session_new();
+  service->cluster = create_cluster(host);
+  if (connect_session(service->session, service->cluster) != CASS_OK) {
+    ta_log_error("%s\n", "connect Scylla cluster fail");
+    return SC_STORAGE_CONNECT_FAIL;
+  }
+  return SC_OK;
+}
+
+status_t db_client_service_free(db_client_service_t* service) {
+  if (service == NULL) {
+    ta_log_error("Invaild input : NULL pointer\n");
+    return SC_TA_NULL;
+  }
+  if (service->session != NULL) {
+    cass_session_free(service->session);
+  }
+  if (service->cluster != NULL) {
+    cass_cluster_free(service->cluster);
+  }
+  return SC_OK;
+}
+
+status_t db_insert_transaction_into_identity_table(db_client_service_t* service, int64_t ID, const char* hash,
```

Comment:
The function name `db_insert_transaction_into_identity_table` is too long. Can you shorten it?  Something like `db_insert_tx_to_idtable` works better.